ui/src/components/pages/Pipeline/partials/Graph/partials/CustomEdge/index.tsx

```diff
@@ -50,6 +50,51 @@ const CustomEdge: FC<EdgeProps> = ({
       labelY = targetY + 2;
     }
   }
+  const getMinWM = () => {
+    if (data?.edgeWatermark?.watermarks) {
+      return Math.min.apply(null, data?.edgeWatermark?.watermarks);
+    } else {
+      return -1;
+    }
+  };
+
+  const getDelay = () => {
+    const str = "Delay - ";
+    const fetchWMTime = data?.edgeWatermark?.WMFetchTime;
+    const minWM = getMinWM();
+    let difference =
+      (fetchWMTime ? fetchWMTime : Date.now()) - (minWM == -1 ? 0 : minWM);
+
+    const years = Math.floor(difference / (1000 * 60 * 60 * 24 * 365));
+    difference -= years * (1000 * 60 * 60 * 24 * 365);
+    const months = Math.floor(difference / (1000 * 60 * 60 * 24 * 30));
+    difference -= months * (1000 * 60 * 60 * 24 * 30);
```

Comment:
Using 30 days as a month is not always precise. Consider using a third-party library like `moment` to calculate time. Something like below:

```
const startTime = minWM === -1 ? 0 : minWM;
const endTime = fetchWMTime || Date.now();

const duration = moment.duration(moment(endTime).diff(moment(startTime)));
const years = duration.years();
const months = duration.months();
...
```

Reply:
It's not precise but third-party libraries also use a constant for number of days in a year, ...
for example here using moment js, the difference between same dates over `leap-year and non-leap` & `two non-leaps years` is not constant ~ (library considers 366 days / yr)
<img width="845" alt="image" src="https://github.com/numaproj/numaflow/assets/49195734/2e92064e-5148-4db8-bfd6-02e316901136">

should we use the current implementation or switch to this library instead?